#7579 feat: Added initial disclaimer component

stuartwilkes opened the pull request:
Relates https://github.com/wellcometrust/corporate/issues/7579

And works with the following additional PRs:

https://github.com/wellcometrust/corporate-d8/pull/383
https://github.com/wellcometrust/corporate-react/pull/548

Adds a disclaimer component containing title, free text and buttons to accept or dismiss (as per ticket above)

Commit:
Added initial disclaimer component

Commit:
Functioning disclaimer

chriddell committed:
#7579 feat: updates Disclaimer component to use updated props

chriddell commented:
@stuartwilkes I've refactored this a little to be more generic, and less tied to the data structure from Drupal. Also added an event handler for the acceptance of the disclaimer.

stuartwilkes commented:
> @stuartwilkes I've refactored this a little to be more generic, and less tied to the data structure from Drupal. Also added an event handler for the acceptance of the disclaimer.

Fab - thanks Chris. I'm just making a tweak to the Drupal para that powers this, so I'll roll any changes in to your amended version

Commit:
Merge branch 'feat/disclaimer-popup-#7579' of github.com:wellcometrust/corporate-components into feat/disclaimer-popup-#7579

# Conflicts:
#	src/components/Disclaimer/Disclaimer.tsx

Commit:
Rename disclaimer to modal, handle blank href

Commit:
Remove overlooked console.log

chriddell committed:
#7579 feat(chore): updates package-lock
